### DISCUSSION TITLE
Import Errors of Upgrading to latest version

### DISCUSSION THREAD

**Discussion by kattt999** [Textualize/textual#2831] [Q&A] [answered]
Hi, we have a project with rich == 12.5.1, textual == 0.1.18. When I tried to upgrade the dependencies to latest versions, I met the problems
`line 92, in __getattr__
    raise ImportError(f"Package 'textual.widgets' has no class '{widget_class}'")
ImportError: Package 'textual.widgets' has no class 'TreeClick'`

Have anyone met this before and how you solved it?

**Comment by TomJGooding** [comment, marked as answer]
Textual has probably had a few breaking changes since v0.1.18!

Your project would likely need pretty significant changes to update to a more up-to-date version of Textual, more than just your error about the missing TreeClick class.

**Comment by davep** [comment]
As @TomJGooding points out above, a lot has changed in Textual since 0.1.18, the most significant changes been [with the release of 0.2.0](https://github.com/Textualize/textual/releases/tag/v0.2.0), where large parts of the API were changed, support for styling with CSS was added, and a bunch of other things. We've now currently on 0.28.1.

In that time [Textual has had extensive documentation added too](https://textual.textualize.io/); I'd suggest perhaps having a read through that (especially the tutorial) to get a feel for how to work with Textual now. As Tom mentions above, moving a Textual application from being based on 0.1.x to anything from 0.2.x onwards will take some work (note though that the API is a lot more stable now),
